pkg/controllers/node/health/controller.go

```diff
@@ -70,6 +78,26 @@ func (c *Controller) Reconcile(ctx context.Context, node *corev1.Node) (reconcil
 		return reconcile.Result{}, nodeutils.IgnoreNodeClaimNotFoundError(err)
 	}
 
+	nodePoolName, found := nodeClaim.Labels[v1.NodePoolLabelKey]
+	if found {
+		nodePoolHealthy, err := c.isNodePoolHealthy(ctx, nodePoolName)
+		if err != nil {
+			return reconcile.Result{}, client.IgnoreNotFound(err)
+		}
+		if !nodePoolHealthy {
+			return reconcile.Result{RequeueAfter: time.Minute}, nil
```

Comment:
I don't think you need a RequeueAfter -- just watch the nodepool and requeue when there is a change to its status

Reply:
Do you mean the watch the nodes here? I removed to just use events we reconcile on